server/segment_rpc.go

```diff
@@ -44,16 +43,24 @@ var errProfile = errors.New("unrecognized encoder profile")
 var errDuration = errors.New("invalid duration")
 var errCapCompat = errors.New("incompatible capabilities")
 
+var dialTimeout = 2 * time.Second
+
 var tlsConfig = &tls.Config{InsecureSkipVerify: true}
 var httpClient = &http.Client{
-	Transport: &http2.Transport{
+	Transport: &http.Transport{
 		TLSClientConfig: tlsConfig,
-		DialTLS: func(network, addr string, cfg *tls.Config) (gonet.Conn, error) {
-			netDialer := &gonet.Dialer{
-				Timeout: 2 * time.Second,
-			}
-			return tls.DialWithDialer(netDialer, network, addr, cfg)
+		DialTLSContext: func(ctx context.Context, network, addr string) (gonet.Conn, error) {
+			cctx, cancel := context.WithTimeout(ctx, dialTimeout)
+			defer cancel()
+
+			tlsDialer := &tls.Dialer{Config: tlsConfig}
```

Comment:
**nit:** Should this be shared among multiple connections or is it ok/advised to instantiate a new one for every dial?

Reply:
I'm not 100% sure, but I haven't seen any information recommending re-using a single dialer or indicating that instantiating a new dialer for each dial could result in inefficiencies. One data point here is that the [code for tls.Dial](https://cs.opensource.google/go/go/+/refs/tags/go1.17.3:src/crypto/tls/tls.go;l=171) actually instantiates a new `net.Dialer` (not `tls.Dialer`, but I imagine both types would have the same considerations) under the hood whenever it is called so that makes me feel like instantiating a new `tls.Dialer` in our case is probably fine.